tests/Feature/MovieTest.php

```diff
@@ -0,0 +1,127 @@
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get('/movies/create')
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_create_a_movie()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get('/movies/create')
+            ->assertStatus(403);
+    }
+
+    /** @test */
+    public function an_admin_can_add_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->postJson('/movies', [
+                'title' => 'Sample Movie',
+                'synopsis' => 'This is a Sample Synopsis',
+                'year' => 2021,
+                'poster' => 'https://via.placeholder.com/600x750.png/00aa33?text=totam',
+                'trailer' => 'http://www.goyette.biz/',
+                'duration' => '190',
+            ])->assertOk();
+
+        $this->assertDatabaseHas('movies', [
+            'title' => 'Sample Movie',
+            'synopsis' => 'This is a Sample Synopsis',
+        ]);
+    }
+
+    /** @test */
+    public function an_admin_can_edit_a_movie()
+    {
+        $movie = Movie::factory()->create();
+        $admin = User::factory()->admin()->create();
+
+        $this->actingAs($admin)
+            ->get("/movies/{$movie->id}/edit")
+            ->assertOk();
+    }
+
+    /** @test */
+    public function a_non_admin_cannot_edit_a_movie()
+    {
+        $movie = Movie::factory()->create();
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->get("/movies/{$movie->id}/edit")
+            ->assertStatus(403);
+    }
+
+    /** @test */
+    public function an_admin_can_update_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+        $movie = Movie::factory()->create();
+
+        $this->actingAs($admin)
+            ->putJson("/movies/{$movie->id}", [
+                'title' => 'Sample Updated Movie',
+                'synopsis' => 'This is a Sample Updated Synopsis',
+                'year' => 2009,
+                'poster' => 'https://via.placeholder.com/600x750.png/00aa33?text=totam',
+                'trailer' => 'http://www.goyette.biz/',
+                'duration' => '190',
+            ])->assertOk();
+
+        $this->assertDatabaseHas('movies', [
+            'title' => 'Sample Updated Movie',
+            'synopsis' => 'This is a Sample Updated Synopsis',
+        ]);
+    }
+
+    /** @test */
+    public function an_admin_can_delete_a_movie()
+    {
+        $admin = User::factory()->admin()->create();
+        $movie = Movie::factory()->create();
+
+        $this->actingAs($admin)
+            ->delete("/movies/{$movie->id}")
+                ->assertOk();
+
+        $this->assertDatabaseMissing('movies', [
+            'id' => $movie->id,
+        ]);
+    }
+}
```

Comment:
We are testing if the admin can create, delete and update and checking if the values are updated in the database. However, what about the user? It is probably a good idea to check if the user is not able to alter things in the database. For example:
- If a user makes a put request to update a movie, it should throw an error, because the user does not have the correct credentials to do so. Then we should check if the database data was not altered. 

That would be the same for creating a movie. What should be the response if a user, that is not an admin, makes a post request to the movies create endpoint? What are you expecting to happen to the DB?

Reply:
That's right Vick, I added those tests now :)